### PR TITLE
sql: mention that CASCADE is unsupported

### PR DESCRIPTION
**Pull request by dt**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8835)

### CODE REVIEW

**Comment by RaduBerinde** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/truncate.go, line 63 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835#-KQ2Uy4O9j7f0BoQpS62:-KQ2Uy4O9j7f0BoQpS63:b-9b4w0x) ([raw file](https://github.com/cockroachdb/cockroach/blob/9126bce5ab38085a5fbe8338546543f278aa8cfa/sql/truncate.go#L63)):*
> ```Go
> 			return nil, err
> 		} else if err = helper.checkAll(nil); err != nil {
> 			if n.DropBehavior == parser.DropCascade {
> ```

why don't we check this at the beginning instead of in this error case?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835#-:-KQ2Uq18dyuNYE-zyTu8:bnfp4nl)*

**Comment by dt** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/truncate.go, line 63 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835#-KQ2Uy4O9j7f0BoQpS62:-KQ2VC_Ga8Zs6l_gXgMy:b3oauth) ([raw file](https://github.com/cockroachdb/cockroach/blob/9126bce5ab38085a5fbe8338546543f278aa8cfa/sql/truncate.go#L63)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> why don't we check this at the beginning instead of in this error case?
</details>
apparently some ORMs just blindly add CASCADE to every statement, so I don't want to fail statements that happen to include it if they otherwise weren't going to hit any errors or FK violations.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835)*

File: sql/truncate.go [hunk 1]
```diff
@@ -59,6 +60,9 @@ func (p *planner) Truncate(n *parser.Truncate) (planNode, error) {
 		if helper, err := makeFKDeleteHelper(p.txn, *tableDesc, fkTables, colMap); err != nil {
 			return nil, err
 		} else if err = helper.checkAll(nil); err != nil {
+			if n.DropBehavior == parser.DropCascade {
+				return nil, util.UnimplementedWithIssueErrorf(8502, "CASCADE not yet supported: %s", err.Error())
```
Comment (tamird) [inline sql/truncate.go]: not need for .Erro()

**Comment by sploiselle** [comment]
Assuming the error is `CASCADE not yet supported`, :lgtm: 

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835#-:-KQ39k4uQ9gJqLVihHa5:baotee2)*

**Comment by dt** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/truncate.go, line 64 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835#-KQ2UScQG-mjRHqi2Klj-r1-64:-KQMcR-Y96Qmn6sMNl6j:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9126bce5ab38085a5fbe8338546543f278aa8cfa/sql/truncate.go#L64)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> not need for .Error()
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8835)*

**Pull request merged**